src/lua/datetime.lua

```diff
@@ -0,0 +1,611 @@
+            nsec = fraction * 1e9
+        else
+            error('only integer values allowed in timestamp '..
+                  'if nsec, usec, or msecs provided', 2)
+        end
+
+        self.epoch = sec_int
+        self.nsec = nsec
+
+        return self
+    end
+
+    local offset0 = self.tzoffset
+    local offset = obj.tzoffset
+    if offset ~= nil then
+        offset = get_timezone(offset)
+        check_range(offset, -720, 720, 'tzoffset')
+    end
+
+    if obj.tz ~= nil then
+        nyi('tz')
+    end
+
+    -- .year, .month, .day
+    if ymd then
+        datetime_ymd_update(self, y or y0, M or M0, d or d0, offset or offset0)
+    end
+
+    -- .hour, .minute, .second
+    if hms then
+        datetime_hms_update(self, h or h0, m or m0, sec or sec0, offset or offset0)
+    end
+
+    self.epoch, self.nsec = normalize_nsec(self.epoch, self.nsec)
+
+    if offset ~= nil then
+        self.tzoffset = offset
+    end
+
+    return self
+end
+
+local function datetime_strftime(self, fmt)
+    local strfmt_sz = 128
+    local buff = ffi.new('char[?]', strfmt_sz)
+    check_date(self, "datetime.strftime()")
+    check_str(fmt, "datetime.strftime()")
+    builtin.datetime_strftime(self, buff, strfmt_sz, fmt)
+    return ffi.string(buff)
+end
+
+local function datetime_format(self, fmt)
+    if fmt ~= nil then
+        return datetime_strftime(self, fmt)
+    else
+        return datetime_tostring(self)
+    end
+end
+
+ffi.metatype(datetime_t, {
+    __tostring = datetime_tostring,
+    __serialize = datetime_serialize,
+    __eq = datetime_eq,
+    __lt = datetime_lt,
+    __le = datetime_le,
+    __index = {
+        epoch = function(self) return self.epoch end,
+        timestamp = function(self) return self.epoch + self.nsec / 1e9 end,
+
+        usec = function(self) return self.nsec / 1e3 end,
+        msec = function(self) return self.nsec / 1e6 end,
+
+        dt = function(self) return local_dt(self) end,
+        year = function(self) return builtin.tnt_dt_year(local_dt(self)) end,
+        month = function(self) return builtin.tnt_dt_month(local_dt(self)) end,
+
+        yday = function(self) return builtin.tnt_dt_doy(local_dt(self)) end,
+        wday = function(self)
+            return ffi.cast('int32_t', builtin.tnt_dt_dow(local_dt(self))) + 1
+        end,
+        day = function(self)
+            return builtin.tnt_dt_dom(local_dt(self))
+        end,
+        hour = function(self) return math_floor((local_secs(self) / 3600) % 24) end,
+        min = function(self) return math_floor((local_secs(self) / 60) % 60) end,
+        sec = function(self) return self.epoch % 60 end,
+
+        format = datetime_format,
+        totable = datetime_totable,
+        set = datetime_set,
+    }
+})
+
+return setmetatable(
+    {
+        new         = datetime_new,
+
+        now         = local_now,
+
+        is_datetime = is_datetime,
+    }, {
+        __call = function(self, ...) return datetime_from(...) end
```

Comment:
The RFC does not say the module should be callable. Lets not complicate stuff and drop this call.

Reply:
This is remnant from original implementation, which here was behaving like `uuid` module. Or `box.cfg`, or `socket`, and many others. It's rather very idiomatic. Why do you consider it's complicating?

Reply:
`box.cfg` is irrelevant - it does not need `require` and you can't do `box()` call. I can give counter-examples: decimal, buffer, fiber, fio, and more. All these examples are modules which can create objects of special type and they do not do `__call`.

> Why do you consider it's complicating

All the code, which wasn't requested by somebody and can be dropped without any internal problems, is complicating. It adds complexity which is not needed. Even if the complexity is small.

Reply:
Ok, I've removed `__call` from module metatable.

```diff
@@ -606,14 +607,8 @@ ffi.metatype(datetime_t, {
     __index = datetime_index,
 })
 
-return setmetatable(
-    {
-        new         = datetime_new,
-
-        now         = datetime_now,
-
-        is_datetime = is_datetime,
-    }, {
-        __call = function(self, ...) return datetime_from(...) end
-    }
-)
+return setmetatable({
+    new         = datetime_new,
+    now         = datetime_now,
+    is_datetime = is_datetime,
+}, {})
```